### PR TITLE
Add support for level argument in any, all

### PR DESCRIPTION
**Pull request by rliu4439**
## What do these changes do?

Refactored code supporting the `level` argument in `count`.

Added support for the `level` argument for:
* `any`
* `all`

## Related issue number

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -446,9 +477,19 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
...
+                    axis=axis, bool_only=False, skipna=skipna, level=level, **kwargs
                 )
+
+            if level is not None:
+                if bool_only is not None:
+                    raise NotImplementedError(
+                        "Option bool_only is not implemented with option level."
+                    )
+
+                kwargs["skipna"] = skipna
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 2]
```diff
@@ -446,9 +477,19 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
...
+                return self._handle_level_agg(axis, level, "any", **kwargs)
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
                return self._handle_level_agg(axis, level, "any", skipna=skipna, **kwargs)
```

File: modin/pandas/base.py [hunk 3]
```diff
@@ -458,11 +499,20 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
             if bool_only:
                 raise ValueError("Axis must be 0 or 1 (got {})".format(axis))
             # Reduce to a scalar if axis is None.
-            result = self._reduce_dimension(
-                self._query_compiler.any(
-                    axis=0, bool_only=bool_only, skipna=skipna, level=level, **kwargs
+
+            if level is not None:
+                kwargs["skipna"] = skipna
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 4]
```diff
@@ -458,11 +499,20 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
             if bool_only:
                 raise ValueError("Axis must be 0 or 1 (got {})".format(axis))
             # Reduce to a scalar if axis is None.
-            result = self._reduce_dimension(
-                self._query_compiler.any(
-                    axis=0, bool_only=bool_only, skipna=skipna, level=level, **kwargs
+
+            if level is not None:
+                kwargs["skipna"] = skipna
+                return self._handle_level_agg(axis, level, "any", **kwargs)
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
                return self._handle_level_agg(axis, level, "any", skipna=skipna, **kwargs)
```

File: modin/pandas/base.py [hunk 5]
```diff
@@ -693,9 +743,7 @@ def count(self, axis=0, level=None, numeric_only=False):
                 # error thrown by pandas
                 raise TypeError("Can only count levels on hierarchical columns.")
 
-            if isinstance(level, str):
-                level = self.axes[axis].names.index(level)
-            return self.groupby(level=level, axis=axis).count()
+            return self._handle_level_agg(axis, level, "count")
```
Comment (devin-petersohn) [inline modin/pandas/base.py]: Please add the keyword arguments here as well.

Comment (rliu4439) [reply modin/pandas/base.py]: Made the changes, thanks for the review!

File: modin/pandas/base.py [hunk 6]
```diff
@@ -409,6 +421,17 @@ def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
                 return data_for_compute.all(
                     axis=axis, bool_only=False, skipna=skipna, level=level, **kwargs
                 )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 7]
```diff
@@ -409,6 +421,17 @@ def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
                 return data_for_compute.all(
                     axis=axis, bool_only=False, skipna=skipna, level=level, **kwargs
                 )
+
+            if level is not None:
+                if bool_only is not None:
+                    raise NotImplementedError(
+                        "Option bool_only is not implemented with option level."
+                    )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 8]
```diff
@@ -418,11 +441,22 @@ def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
             if bool_only:
                 raise ValueError("Axis must be 0 or 1 (got {})".format(axis))
             # Reduce to a scalar if axis is None.
-            result = self._reduce_dimension(
-                self._query_compiler.all(
-                    axis=0, bool_only=bool_only, skipna=skipna, level=level, **kwargs
+
+            if level is not None:
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 9]
```diff
@@ -418,11 +441,22 @@ def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
             if bool_only:
                 raise ValueError("Axis must be 0 or 1 (got {})".format(axis))
             # Reduce to a scalar if axis is None.
-            result = self._reduce_dimension(
-                self._query_compiler.all(
-                    axis=0, bool_only=bool_only, skipna=skipna, level=level, **kwargs
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 10]
```diff
@@ -409,6 +421,17 @@ def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
...
+            if level is not None:
+                if bool_only is not None:
+                    raise NotImplementedError(
+                        "Option bool_only is not implemented with option level."
+                    )
+
+                return self._handle_level_agg(
+                    axis, level, "all", skipna=skipna, **kwargs
+                )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 11]
```diff
@@ -446,9 +480,20 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
                         )
                     )
                 data_for_compute = self[self.columns[self.dtypes == np.bool]]
-                return data_for_compute.all(
-                    axis=axis, bool_only=None, skipna=skipna, level=level, **kwargs
+                return data_for_compute.any(
+                    axis=axis, bool_only=False, skipna=skipna, level=level, **kwargs
+                )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 12]
```diff
@@ -446,9 +480,20 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
...
+                return data_for_compute.any(
+                    axis=axis, bool_only=False, skipna=skipna, level=level, **kwargs
+                )
+
+            if level is not None:
+                if bool_only is not None:
+                    raise NotImplementedError(
+                        "Option bool_only is not implemented with option level."
+                    )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 13]
```diff
@@ -446,9 +480,20 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
...
+                return self._handle_level_agg(
+                    axis, level, "any", skipna=skipna, **kwargs
                 )
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

File: modin/pandas/base.py [hunk 14]
```diff
@@ -458,11 +503,21 @@ def any(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
             if bool_only:
                 raise ValueError("Axis must be 0 or 1 (got {})".format(axis))
             # Reduce to a scalar if axis is None.
-            result = self._reduce_dimension(
-                self._query_compiler.any(
-                    axis=0, bool_only=bool_only, skipna=skipna, level=level, **kwargs
+
```
Comment (devin-petersohn) [inline modin/pandas/base.py]:
```suggestion
```

**Comment by devin-petersohn** [comment]
@rliu4439 I do not have the privileges to add these. Can you add these suggested changes? You can do it in the browser.

**Pull request merged**